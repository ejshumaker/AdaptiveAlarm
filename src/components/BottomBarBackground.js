import React from 'react';
import {
  View, StyleSheet, Platform,
} from 'react-native';

// const { StatusBarManager } = NativeModules;

const styles = StyleSheet.create({
  bottomBarBackground: {
    paddingVertical: (Platform.OS === 'ios') ? 96 : 96,
    backgroundColor: 'rgba(255,255,255,0)',
  },
});

const BottomBarBackground = (props) => {
  // eslint-disable-next-line
  const { style } = props;
  return (
    <View style={[styles.bottomBarBackground, style || {}]} />
  );
};


export default BottomBarBackground;
