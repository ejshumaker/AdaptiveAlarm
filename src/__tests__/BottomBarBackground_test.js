/* eslint-disable no-undef */
/* eslint-disable import/first */
import 'react-native';
import React from 'react';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import BottomBarBackground from '../components/BottomBarBackground';

// jest.mock('react-native-svg');

Enzyme.configure({ adapter: new Adapter() });

jest.setTimeout(10000);
let wrapper = null;
const onPressMock = jest.fn();


describe('Bottom Bar tests', () => {
  beforeEach(() => {
    wrapper = mount(<BottomBarBackground
      onPress={onPressMock}
      title="test"
      textColor="b"
      backgroundColor="w"
    />);
    jest.clearAllMocks();
  });

  it('test BottomBarBackground matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
