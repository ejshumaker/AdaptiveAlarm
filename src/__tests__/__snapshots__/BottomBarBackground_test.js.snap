// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bottom Bar tests test BottomBarBackground matches snapshot 1`] = `
<BottomBarBackground
  backgroundColor="w"
  onPress={[MockFunction]}
  textColor="b"
  title="test"
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "rgba(255,255,255,0)",
          "paddingVertical": 96,
        },
        Object {},
      ]
    }
  >
    <react-native-mock />
  </View>
</BottomBarBackground>
`;

exports[`Buttons tests test BottomBarBackground matches snapshot 1`] = `
<BottomBarBackground
  backgroundColor="w"
  onPress={[MockFunction]}
  textColor="b"
  title="test"
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "rgba(255,255,255,0)",
          "paddingVertical": 96,
        },
        Object {},
      ]
    }
  >
    <react-native-mock />
  </View>
</BottomBarBackground>
`;
